src/modules/deal-guarantee/deal-guarantee.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@nestjs/common';
+import { PROPERTIES } from '@ukef/constants';
+import { AcbsAuthenticationService } from '@ukef/modules/acbs/acbs-authentication.service';
+import { AcbsDealGuaranteeService } from '@ukef/modules/acbs/acbs-deal-guarantee.service';
+import { AcbsCreateDealGuaranteeDto } from '@ukef/modules/acbs/dto/acbs-create-deal-guarantee.dto';
+import { CurrentDateProvider } from '@ukef/modules/date/current-date.provider';
+
+import { DealGuaranteeToCreate } from './deal-guarantee-to-create.interface';
+
+@Injectable()
+export class DealGuaranteeService {
+  constructor(
+    private readonly acbsAuthenticationService: AcbsAuthenticationService,
+    private readonly acbsDealGuaranteeService: AcbsDealGuaranteeService,
+    private readonly currentDateProvider: CurrentDateProvider,
+  ) {}
+
+  async createGuaranteeForDeal(dealIdentifier: string, newGuarantee: DealGuaranteeToCreate): Promise<void> {
+    const idToken = await this.acbsAuthenticationService.getIdToken();
+
+    const effectiveDateTime = this.currentDateProvider.getLatestDateFromTodayAnd(new Date(newGuarantee.effectiveDate + 'T00:00:00Z'));
+    const effectiveDateOnlyString = effectiveDateTime.toISOString().split('T')[0];
+
+    const guaranteeToCreateInAcbs: AcbsCreateDealGuaranteeDto = {
+      LenderType: {
+        LenderTypeCode: PROPERTIES.DEAL_GUARANTEE.DEFAULT.lenderType.lenderTypeCode,
+      },
+      SectionIdentifier: PROPERTIES.DEAL_GUARANTEE.DEFAULT.sectionIdentifier,
+      LimitType: {
+        LimitTypeCode: PROPERTIES.DEAL_GUARANTEE.DEFAULT.limitType.limitTypeCode,
+      },
+      LimitKey: newGuarantee.limitKey,
+      GuarantorParty: {
+        PartyIdentifier: newGuarantee.guarantorParty ?? PROPERTIES.DEAL_GUARANTEE.DEFAULT.guarantorParty,
+      },
+      GuaranteeType: {
+        GuaranteeTypeCode: newGuarantee.guaranteeTypeCode ?? PROPERTIES.DEAL_GUARANTEE.DEFAULT.guaranteeTypeCode,
+      },
+      EffectiveDate: effectiveDateOnlyString + 'T00:00:00Z',
+      ExpirationDate: newGuarantee.guaranteeExpiryDate + 'T00:00:00Z',
+      GuaranteedLimit: Math.round(newGuarantee.maximumLiability * 100) / 100, // TODO APIM-73: Discuss rounding precision errors with team
```

Comment:
Mulesoft rounds this value to 2dp, so I looked for a way to do this in JS. Unfortunately, all of the options will suffer from floating point precision errors. Is this the same in Mulesoft @avaitonis?

Reply:
Mulesoft documentation says that precision is not lost in operations, but we didn't test float operations. 

We don't use very big or very small values, does float precision affect us? how to replicate this issue?

Here is some tests in Mulesoft:
![image](https://user-images.githubusercontent.com/2437608/225940454-84c3ae6f-ce78-4e76-82c7-be8ca22d941d.png)

Reply:
When rounding, there is an issue with float representations of certain decimal values. For example, you'd expect `1.005` to round to `1.01` but my code will actually round it to `1.00` (or really just `1`) because `1.005` = `1.0049999999999999` in the float representation

Reply:
Can we use Math.round((newGuarantee.maximumLiability + Number.EPSILON) * 100) / 100).
And move it to helper.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/EPSILON

Code rounds 1.0049999999999997 to 1 and 1.0049999999999998 to 1.005